replicate/common/generic-replicator.go

```diff
@@ -117,7 +117,7 @@ func (r *GenericReplicator) IsReplicationPermitted(object *metav1.ObjectMeta, so
 	allowedNamespaces := strings.Split(annotationAllowedNamespaces, ",")
 	allowed := false
 	for _, ns := range allowedNamespaces {
-		ns := strings.TrimSpace(ns)
+		ns := fmt.Sprintf("^%s$", strings.TrimSpace(ns))
```

Comment:
Just a quick thought: What if someone _already_ specified a regex with start and/or end markers in their annotation (e.g. `replicator.v1.mittwald.de/replicate-to: "^my-ns-1$"`)? In this case, this would compile to a regex like `^^my-ns-1$$`. Maybe we should add a workaround for this?

```go
s = strings.TrimSpace(s)

if !strings.HasPrefix(s, "^") {
    s = "^" + s
}

if !strings.HasSuffix(s, "$") {
    s = s + "$"
}
```

❓

Reply:
From testing on go playground it doesn't look like it causes issues, i.e. ```^^my-ns-1$$``` still matches ```my-ns-1```, but I'll add it to the dedicated regex-building function to be on the safe side